service/controller/v5/resource/vpn/resource.go

```diff
@@ -23,15 +23,17 @@ type Config struct {
 	Debugger *debugger.Debugger
 	Logger   micrologger.Logger
 
-	Azure setting.Azure
+	Azure           setting.Azure
+	TemplateVersion string
 }
 
 // Resource ensures Microsoft Virtual Network Gateways are running.
```

Comment:
In general I think it would be good to have a godoc explaining why we have a separate resource for the gateway. I.e. that for w/e reason it blocks machines creation and to speed up things we create a separate deployment to do it in parallel.

It's fine to do that in separate PR.